### ISSUE TITLE
Store in other folders apart from cache dir?

### ISSUE THREAD

**Issue by IODevBlue** [rommansabbir/StoreX#4]
Great library!

I noticed internally it stores objects in application's cache directory (cacheDir). Is it possible to provide an api to store in other custom folders?

**Comment by rommansabbir** [comment]
Yes, in that case, client can provide their desired __path__ for storing __StoreAbleObject__. So, all object will be stored under __/data/user/0/package_name/files/custom_path_by_client/testObject__. Is this what you want to be inclued in future version?

**Comment by IODevBlue** [comment]
Yes!

I planned to store encrypted objects in different folder paths in the sdcard or in my app's sdcard cache. I had several entities which I needed to store in different file paths but there's no way to do that. It seems StoreX caches all entities in the same folder.
Eg.
For user profiles:
   /sdcard0/Android/data/package_name/cache/profile/

For recently viewed profiles:
 /sdcard0/Android/data/package_name/cache/user/user/recent/

Also, I did notice these:
- StoreXCore has the encryption_key variable set as a public var (which automatically generates a setter) and there is also a setEncryptionKey() method which does the actual check on the encryption key to be set and throws the InvalidEncryptionKeyException if key is empty. The public var has rendered the setEncryptionKey() redundant as the key can be set without the proper check.
- In StoreX, the put() method with coroutine support has no return Boolean unlike the function it replaces (line 16):
> fun put(scope: CoroutineScope, key: String, value: StoreAbleObject) 

The deprecated function has a return Boolean (line 13):
> fun put(key: String, value: StoreAbleObject): Boolean

- BaseStoreXInstance only clears the cached objects in SharedPreference but not those stored as files in both clearCacheByKey() and clearAllCache(). I assume that cached objects as files would still remain in storage even though they might be redundant. (Line 86)
> override fun clearCacheByKey(key: String) {
        mSharedPreferences.edit().remove(key).apply()
    }

(Line 90):
> override fun clearAllCache() {
        mSharedPreferences.edit().clear().apply()
    }
